Backpack-SwiftUI/SponsoredBanner/Classes/BPKSponsoredBanner.swift

```diff
@@ -0,0 +1,171 @@
+//
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright © 2023 Skyscanner Ltd. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.a=pache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import SwiftUI
+
+public struct BPKSponsoredBanner: View {
+    let logo: Image?
+    let title: String?
+    let subheadline: String?
+    let callToAction: CallToAction?
+    let bodyText: String?
+    let variant: Variant
+    let backgroundColor: Color
+    
+    @State var showTerms: Bool
+    
+    public init(
+        logo: Image? = nil,
+        title: String? = nil,
+        subheadline: String? = nil,
+        callToAction: CallToAction? = nil,
+        bodyText: String? = nil,
+        variant: Variant,
+        backgroundColor: Color,
+        showTerms: Bool = false
+    ) {
+        self.logo = logo
+        self.title = title
+        self.subheadline = subheadline
+        self.callToAction = callToAction
+        self.bodyText = bodyText
+        self.variant = variant
+        self.backgroundColor = backgroundColor
+        self.showTerms = showTerms
+    }
+    
+    public var body: some View {
+        VStack(spacing: 0) {
+            topView
+                .padding(.base)
+                .background(backgroundColor)
+                .zIndex(1)
+                .onTapGesture { toggleBodyText() }
+            if let bodyText = bodyText, showTerms {
+                BPKText(bodyText, style: .caption)
+                    .lineLimit(4)
+                    .frame(maxWidth: .infinity)
+                    .padding(.base)
+                    .background(Color.clear)
+                    .zIndex(0)
+            }
+        }
+        .background(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+                .foregroundColor(.canvasContrastColor)
+        )
+        .clipShape(
+            RoundedRectangle(cornerRadius: .sm, style: .continuous)
+        )
+    }
+    
+    private var topView: some View {
+        HStack(spacing: .md) {
+            HStack {
```

Comment:
Why do we need 2 Hstack levels here?

Reply:
Good catch, initially it was to have separation of the info label from the rest, but perhaps it's not needed anymore 🤔

Reply:
Done, check 98da5d5e583a83f8c6ac8ddbf181a0c07f300667